app/components/spina/media_picker/modal_component.html.erb

```diff
@@ -58,17 +58,7 @@
               <% end %>
             <% end %>
             
-            <%= form_with model: [:admin, Spina::Image.new], url: helpers.spina.admin_images_path, data: {controller: "form loading-button", loading_message: t('spina.media_library.uploading'), action: "turbo:submit-end->loading-button#doneLoading", turbo_frame: "media_picker"} do |f| %>
-              <%= hidden_field_tag :modal, true %>
-              <%= f.hidden_field :media_folder_id, value: @media_folder&.id %>
-              
-              <%= f.file_field :files, multiple: true, accept: "image/*", id: "new_image_file_field", class: 'hidden', data: {action: "loading-button#loading form#requestSubmit"} %>
-              
-              <button type="button" class="font-medium w-full text-gray-600 text-sm hover:bg-gray-200 px-3 py-2 cursor-pointer rounded-lg flex items-center w-full" data-controller="delegate-click" data-action="delegate-click#click" data-loading-button-target="button" data-delegate-click-target="#new_image_file_field">
-                <%= helpers.heroicon("upload", style: :solid, class: "w-5 h-5 text-spina mr-2") %>
-                <%=t 'spina.media_library.upload_from_device' %>
-              </button>
-            <% end %>
+            <%= render Spina::Forms::FileUploadComponent.new(origin: 'media-picker', media_folder: @media_folder, turbo_frame: 'media_picker') %>
```

Comment:
Refactored the media picker file upload functionality into its own component, which is reused by the automatic file uploader. There is similar code on the media library page which could be refactored to use this component as well.

Reply:
The only thing getting in the way of converting the upload form on `app/views/spina/admin/images/index.html.erb` so that it uses the same component is the styling of the button, which I'd either have to make consistent or do a block to allow dependency injection for customization of its appearance.
![image](https://github.com/SpinaCMS/Spina/assets/1912237/dc945aad-f25a-47c9-8e2f-b1396637c6ee)
![image](https://github.com/SpinaCMS/Spina/assets/1912237/9ca6fd4b-d6d0-4dd9-a1c7-41c4de45ed71)